src/js/versions/version_controller.js

```diff
@@ -304,37 +314,58 @@ export class VersionController extends Component {
     loadAllVersions(versions) {
         let self = this;
         self.versions = versions;
-        if (self._activeVersion !== null) {
-            this.setActiveVersionByName(self._activeVersion.model.name);
-        } else {
-            self._activeVersion = self.versions.filter((v) => {
-                return v.model.isDefault
-            })[0];
-        }
 
         self.versions.forEach((version) => {
             self.getAllDetails(version);
         })
 
         Promise.all(this._promises).then(() => {
             //populate rich data, data mapper..
+            self.checkAndSwitchVersion();
             this._promises = [];
             this.parent.triggerEvent(VersionController.EVENTS.ready, this.allVersionsBundle);
         });
     }
 
+    checkAndSwitchVersion() {
+        // if the geography does not exist in the default version, switch to another version
+        const self = this;
+
+        if (self._activeVersion === null) {
+            self._activeVersion = self.versions.filter((v) => {
+                return v.model.isDefault && v.model.exists
+            })[0];
+        }
+        if (self._activeVersion === null || self._activeVersion === undefined || !self._activeVersion.model.exists) {
+            self._activeVersion = self.versions.filter((v) => {
+                return v.model.exists
+            })[0];
+        }
+
+        if (self._activeVersion === null || self._activeVersion === undefined) {
+            throw 'Selected geography does not exist in any of the versions'
+        }
```

Comment:
The if statements don't seem right to me e.g. In a case where `self._activeVersion === null` It will can only enter the first statement. The next if clauses will not be reached. Similar case for `self._activeVersion === undefined`

Reply:
* first `if` statement : if `self._activeVersion` is null, assign the version that is marked as default and is not 404
* second `if` statement : if `self._activeVersion` is still null, it means the default version is 404. so just assign the first version that is not 404
* third `if` statement : the selected geography does not exist in any of the versions. throw an error.

notice that those are not `else if`s - they are seperate `if`s. I don't see a problem here

Reply:
if you want to make sure, you can debug this code by navigating to geo:87ad36151ffffff in sandbox YE profile. since the default version is 404, you'd see that the second `if` is hit. 

if you go to a geography which does not exist like geo:random123 you will see the third `if` is hit